tests/test_users.py

```diff
@@ -0,0 +1,232 @@
+        token = self.login().json['token']
+        all_us = self.client.get('/api/users', headers={'token': token})
+        user = json.loads(all_us.data)[0]['id']
+        token = self.login().json['token'] + '1'
+        single_user = self.client.get('/api/users/' + str(user),
+                                      headers={'token': token})
+        self.assertEqual(single_user.status_code, 401)
+
+    def test_get_single_user_with_no_id(self):
+        token = self.login().json['token']
+        all_us = self.client.get('/api/users', headers={'token': token})
+        single_user = self.client.get('/api/users/', headers={'token': token})
+        self.assertEqual(single_user.status_code, 404)
+
+    def test_get_single_user_with_invalid_id(self):
+        token = self.login().json['token']
+        all_us = self.client.get('/api/users', headers={'token': token})
+        user = json.loads(all_us.data)
+        for item in user:
+            max_id = 1
+            if item['id'] > max_id:
+                max_id = item['id']
+        single_user = self.client.get('/api/users/' + str(max_id + 1),
+                                      headers={'token': token})
+        self.assertEqual(single_user.status_code, 404)
+
+    def test_get_single_user(self):
+        token = self.login().json['token']
+        all_us = self.client.get('/api/users', headers={'token': token})
+        user = json.loads(all_us.data)[0]['id']
+        single_user = self.client.get('/api/users/' + str(user),
+                                      headers={'token': token})
+        self.assertEqual(single_user.status_code, 200)
+
+
+class TestDeleteUser(BaseTestCase):
+
+    def test_delete_single_user_with_no_token(self):
+        token = self.login().json['token']
+        all_us = self.client.get('/api/users', headers={'token': token})
+        user = json.loads(all_us.data)[0]['id']
+        single_user = self.client.delete('/api/users/' + str(user))
+        self.assertEqual(single_user.status_code, 401)
+
+    def test_delete_single_user_with_invalid_token(self):
+        token = self.login().json['token']
+        all_us = self.client.get('/api/users', headers={'token': token})
+        user = json.loads(all_us.data)[0]['id']
+        token = self.login().json['token'] + '1'
+        single_user = self.client.delete('/api/users/' + str(user),
+                                         headers={'token': token})
+        self.assertEqual(single_user.status_code, 401)
+
+    def test_delete_single_user_with_no_id(self):
+        token = self.login().json['token']
+        all_us = self.client.get('/api/users', headers={'token': token})
+        single_user = self.client.get('/api/users/', headers={'token': token})
+        self.assertEqual(single_user.status_code, 404)
+
+    def test_delete_single_user_with_invalid_id(self):
+        token = self.login().json['token']
+        all_us = self.client.get('/api/users', headers={'token': token})
```

Comment:
at this point you already now how many users you have, so it is not needed to call api method and find max user id.
you could use hardcoded value like 100.